src/m365/spo/commands/site/site-commsite-enable.ts

```diff
@@ -67,33 +64,25 @@ class SpoSiteCommSiteEnableCommand extends SpoCommand {
   }
 
   public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
-    const designPackageId: string = args.options.designPackageId || '{d604dac3-50d3-405e-9ab9-d4713cda74ef}';
+    const designPackageId: string = args.options.designPackageId || '96c933ac-3698-44c7-9f4a-5fd17d71af9e';
 
-    try {
-      const spoAdminUrl: string = await spo.getSpoAdminUrl(logger, this.debug);
-
-      if (this.debug) {
-        logger.logToStderr(`Retrieving request digest...`);
-      }
+    if (this.verbose) {
+      logger.logToStderr(`Enabling communication site at ${args.options.url}...`);
+    }
 
-      const ctxInfo: ContextInfo = await spo.getRequestDigest(spoAdminUrl);
-      const requestOptions: any = {
-        url: `${spoAdminUrl}/_vti_bin/client.svc/ProcessQuery`,
+    try {
+      const requestOptions: CliRequestOptions = {
+        url: `${args.options.url}/_api/sitepages/communicationsite/enable`,
         headers: {
-          'X-RequestDigest': ctxInfo.FormDigestValue
+          'accept': 'application/json;odata=nometadata'
         },
-        data: `<Request AddExpandoFieldTypeSuffix="true" SchemaVersion="15.0.0.0" LibraryVersion="16.0.0.0" ApplicationName="${config.applicationName}" xmlns="http://schemas.microsoft.com/sharepoint/clientquery/2009"><Actions><ObjectPath Id="4" ObjectPathId="3" /><Method Name="EnableCommSite" Id="5" ObjectPathId="3"><Parameters><Parameter Type="String">${formatting.escapeXml(args.options.url)}</Parameter><Parameter Type="Guid">${formatting.escapeXml(designPackageId)}</Parameter></Parameters></Method></Actions><ObjectPaths><Constructor Id="3" TypeId="{268004ae-ef6b-4e9b-8425-127220d84719}" /></ObjectPaths></Request>`
+        data: { designPackageId }
       };
 
-      const res: string = await request.post(requestOptions);
-      const json: ClientSvcResponse = JSON.parse(res);
-      const response: ClientSvcResponseContents = json[0];
-      if (response.ErrorInfo) {
-        throw response.ErrorInfo.ErrorMessage;
-      }
-    } 
+      await request.post(requestOptions);
+    }
     catch (err: any) {
-      this.handleRejectedODataJsonPromise(err);
+      this.handleRejectedODataPromise(err);
```

Comment:
This works indeed. However we are expecting a `json` response from the API call. Therefore I suggest that we add the `responseType` parameter to the `requestOptions` and put this back to `handleRejectedODataJsonPromise`.